### PR TITLE
Add support for SUNION and SINTER commands

### PR DESCRIPTION
**Pull request by aburakovskiy**
Add support for commands:
`SUNION` https://redis.io/commands/sunion
`SINTER`  https://redis.io/commands/sinter

### CODE REVIEW

**Comment by aburakovskiy** [comment]
@Fabex do you mean do not use `smembers` command inside?

**Comment by Fabex** [comment]
No I just ask you to remove else instructions.
You do, if something set keys else set keys.
You can just replace the if else by a ternary or by a default set

File: src/M6Web/Component/RedisMock/RedisMock.php [hunk 1]
```diff
@@ -308,6 +308,43 @@ public function smembers($key)
         return $this->returnPipedInfo(self::$dataValues[$this->storage][$key]);
     }
 
+    public function sunion($key)
+    {
+        $this->stopPipeline();
+        if (is_array($key)) {
+            $keys = $key;
+        } else {
```
Comment (Fabex) [inline src/M6Web/Component/RedisMock/RedisMock.php]: to be more explicit here a example:
```php
$keys = is_array($key) ? $key : func_get_args();
```

File: src/M6Web/Component/RedisMock/RedisMock.php [hunk 2]
```diff
@@ -308,6 +308,43 @@ public function smembers($key)
...
+
+        $this->restorePipeline();
+
+        return $this->returnPipedInfo($result);
+    }
+
+    public function sinter($key)
+    {
+        $this->stopPipeline();
+        if (is_array($key)) {
```
Comment (Fabex) [inline src/M6Web/Component/RedisMock/RedisMock.php]: same here

```php
$keys = is_array($key) ? $key : func_get_args();
```

File: src/M6Web/Component/RedisMock/RedisMock.php [hunk 3]
```diff
@@ -308,6 +308,35 @@ public function smembers($key)
         return $this->returnPipedInfo(self::$dataValues[$this->storage][$key]);
     }
 
+    public function sunion($key)
+    {
+        $this->stopPipeline();
+        $keys = is_array($key) ? $key : func_get_args();
+        $result = array();
```
Comment (Fabex) [inline src/M6Web/Component/RedisMock/RedisMock.php]: just one last thing
can you use short array syntax please.

$result = [];

File: src/M6Web/Component/RedisMock/RedisMock.php [hunk 4]
```diff
@@ -308,6 +308,35 @@ public function smembers($key)
...
+        $this->restorePipeline();
+
+        return $this->returnPipedInfo($result);
+    }
+
+    public function sinter($key)
+    {
+        $this->stopPipeline();
+        $keys = is_array($key) ? $key : func_get_args();
+        $result = array();
```
Comment (Fabex) [inline src/M6Web/Component/RedisMock/RedisMock.php]: same review

**Comment by omansour** [comment]
v3.1.0

txs @aburakovskiy

**Pull request merged**